spec/unit/rack/user_spec.rb

```diff
@@ -65,13 +66,28 @@ def env_for(url, opts = {})
 
       context "and it is not nil" do
         it "returns the wrapped user" do
-          controller = instance_double('DummyController')
-          env = env_for('/', 'action_controller.instance' => controller)
           allow(controller).to receive(:current_user) { user }
 
           retval = described_class.extract(env)
           expect(retval).to be_a(described_class)
         end
+
+        context "but it requires parameters" do
+          before do
+            class DummyController
+              def current_user(_)
```

Comment:
Please add tests that cover:

* `def current_user(a)`
* `def current_user(a, b)`
* `def current_user(*a)`
* `def current_user(a, *b)`
* `def current_user(a, b, *c, &d)`